src/tmpfiles/tmpfiles.c

```diff
@@ -2003,6 +2005,177 @@ static int glob_item_recursively(Item *i, fdaction_t action) {
         return r;
 }
 
+static int rm_if_wrong_type_safe(
+                mode_t mode,
+                int parent_fd,
+                const struct stat *parent_st /* Only used if follow is true. */,
+                const char *name,
+                bool follow) {
```

Comment:
instead of a "bool follow" i'd do "int flags" which you then pass directly to fstatat() and friends. it's usually easier to read function invocation then, since flag names say what is meant, while just "true" or "false" does not.